frontend/cmd/cmd.go

```diff
@@ -168,18 +170,33 @@ func (opts *FrontendOpts) Run() error {
 	}
 	logger.Info(fmt.Sprintf("Application running in %s", opts.location))
 
-	f := frontend.NewFrontend(logger, listener, metricsListener, prometheusEmitter, dbClient, opts.location, &csClient)
-
 	stop := make(chan struct{})
 	signalChannel := make(chan os.Signal, 1)
 	signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
-	go f.Run(ctx, stop)
+
+	var wg sync.WaitGroup
+
+	f := frontend.NewFrontend(logger, listener, metricsListener, prometheusEmitter, dbClient, opts.location, &csClient)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		f.Run(ctx, stop)
+	}()
+
+	collector := metrics.NewSubscriptionCollector(prometheus.DefaultRegisterer, dbClient, opts.location)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		collector.Run(logger, stop)
+	}()
```

Comment:
Could this be moved inside `Frontend.Run`?  We're already using a wait group (actually an [error group](https://pkg.go.dev/golang.org/x/sync/errgroup)) there.  Just modify `SubscriptionCollector.Run` to return an `error`.

Reply:
much simper indeed!